core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,310 @@
+	// RateLimiter    *values.Map `toml:"rateLimiter"`
+	RequiredParams []string `toml:"requiredParams"`
+}
+
+// TODO: Question asked in Web API Trigger about this structure
+// https://docs.google.com/document/d/1mCTAo-ix-P923eUlh4SloZfBN9PCvgf90oHWbmykjsc/edit?disco=AAABWF65hAM
+// Eventually can have PENDING or COMPLETED so status is different than Success
+
+type Response struct {
+	Success      bool   `json:"success"`
+	ErrorMessage string `json:"error_message,omitempty"`
+	Status       string `json:"status"`
+}
+
+type webapiTrigger struct {
+	allowedSendersMap map[string]bool
+	allowedTopicsMap  map[string]bool
+	ch                chan<- capabilities.TriggerResponse
+	config            TriggerConfig
+	// rateLimiter       *common.RateLimiter
+}
+
+type triggerConnectorHandler struct {
+	services.StateMachine
+
+	capabilities.CapabilityInfo
+	capabilities.Validator[TriggerConfig, Input, capabilities.TriggerResponse]
+	connector           connector.GatewayConnector
+	lggr                logger.Logger
+	mu                  sync.Mutex
+	registeredWorkflows map[string]webapiTrigger
+	signerKey           *ecdsa.PrivateKey
+}
+
+var _ capabilities.TriggerCapability = (*triggerConnectorHandler)(nil)
+var _ services.Service = &triggerConnectorHandler{}
+
+func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, signerKey *ecdsa.PrivateKey, lggr logger.Logger) (*triggerConnectorHandler, error) {
+	handler := &triggerConnectorHandler{
+		Validator:           capabilities.NewValidator[TriggerConfig, Input, capabilities.TriggerResponse](capabilities.ValidatorArgs{Info: webapiTriggerInfo}),
+		connector:           connector,
+		signerKey:           signerKey,
+		registeredWorkflows: map[string]webapiTrigger{},
+		lggr:                lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	return handler, nil
+}
+
+// Iterate over each topic, checking against senders and rateLimits, then starting event processing and responding
+func (h *triggerConnectorHandler) processTrigger(ctx context.Context, gatewayID string, body *api.MessageBody, sender ethCommon.Address, payload workflow.TriggerRequestPayload) {
+	wrappedPayload, _ := values.WrapMap(payload)
+
+	for _, trigger := range h.registeredWorkflows {
+		topics := payload.Topics
+
+		for _, topic := range topics {
+			if !trigger.allowedTopicsMap[topic] {
+				return
+			}
+		}
+
+		if !trigger.allowedSendersMap[sender.String()] {
+			h.lggr.Errorw("Unauthorized Sender")
+			return
+		}
+		// if !trigger.rateLimiter.Allow(body.Sender) {
+		// 	h.lggr.Errorw("request rate-limited")
+		// 	return
+		// }
+
+		TriggerEventID := body.Sender + payload.TriggerEventID
+		tr := capabilities.TriggerResponse{
+			Event: capabilities.TriggerEvent{
+				TriggerType: triggerType,
+				ID:          TriggerEventID,
+				Outputs:     wrappedPayload,
+			},
+		}
+
+		trigger.ch <- tr
+
+		response := Response{Success: true, Status: "ACCEPTED"}
```

Comment:
May be able to simplify this to just Status, and have the Gateway switch case between them